### ISSUE TITLE
BUG: transaction preparation fails for non-account-deploying transactions on custom networks

### ISSUE THREAD

**Issue by xJonathanLEI** [argentlabs/argent-x#2090]
## Summary

The extension throws a "RPC is not supported for v4 deprecated provider" error when using a local `katana` network on any transaction other than the one that bundles with account deployment.

## Steps to reproduce

I'm hosting a temporary `katana` RPC endpoint until this issue is resolved.

> [!NOTE]
>
> You can technically use your own `katana` instance, but the out-of-the-box `katana` installation doesn't quite work with Argent X. You will need to:
>
> 1. Declare and deploy the `Multicall` class
> 2. Declare the Argent X account contract class
>
> Or just use the instance I'm hosting (if it's still available when you read this).

1. Create a network with the following details:
   - Network name: Katana
   - Chain ID: KATANA
   - RPC: https://rpc-katana-temp.starknet.rs/
2. Create an account for this network
3. Fund this account by sending ETH from the default prefunded accounts in katana

   An example Starkli command to send yourself 10 ETH:

   ```console
   starkli invoke --rpc "https://rpc-katana-temp.starknet.rs/" --account katana-0 eth transfer 0xYOUR_ACCOUNT u256:10000000000000000000
   ```

4. Attempt to make any transaction from this account (e.g. ETH transfer) - it will deploy the acount and make the transaction just fine
5. Attempt to make any transaction again

## Expected result

The second transaction should also be prepared successfully.

## Actual result

<img width="356" alt="image" src="https://github.com/argentlabs/argent-x/assets/19556359/7657ab3c-897c-4950-9ac0-f334e50e396d">

Upon checking the background page console:

```log
SIMULATE_TRANSACTIONS_REJ Error: RPC is not supported for v4 deprecated provider
    at B (background.js:38:100973)
    at Rm.getStarknetAccount (background.js:2231:66167)
    at async async.msg (background.js:2248:46554)
    at async Object.hp [as next] (background.js:2248:49001)
async.msg @ background.js:2248
```

**Comment by Cevedale** [comment]
Update the extension or the library that manages the Starknet provider to a version that supports the RPC operation.  Version (v4) is deprecated and not suitable for the RPC call. Consult the official documentation or release notes for the extension or library to find the latest version that supports RPC.

Check if the Starkli command-line tool is up to date. You can update it using the following command:

bash
Copy code
npm install -g starkware-cli
This assumes you have Node.js and npm installed. If not, you can install them from https://nodejs.org/en .

Verify that the network configuration (Katana) is correctly set up. Ensure that the RPC endpoint is accessible and the network details (chain ID, RPC URL) match the actual configuration of the Katana network. Sometimes, these details can change, and using outdated configurations can lead to issues.

Don't forget to update also the Starknet extension.

**Comment by xJonathanLEI** [comment]
Damn is this an AI generated comment?

In any case, I believe this is fixed in v5.10.0. I just tested and it works now. Closing this issue.
